theme/layouts/_navigation.ejs

```diff
@@ -1,22 +1,54 @@
-<nav id="main-navigation">
+<%
+const query = site.queryPages;
+
+function getSubPages(nav)
+{
+  const subPages = query(({originalPathname}) => {
+    const parentPath = `${nav.originalPathname}/`;
+    return originalPathname.startsWith(parentPath) &&
+           !(originalPathname.substring(parentPath.length).includes("/"));
+  });
+  if (subPages.length) {
+    if (!nav.sub_items){
+      nav.sub_items = [];
+      for (const subPage of subPages) {
+        if(subPage.menu !== false) {
+        nav.sub_items.push(subPage);
+        getSubPages(subPage);
+        }
+      }
+      nav.sub_items.sort((a, b) => (a.weight > b.weight) ? 1 : -1);
```

Comment:
> In order to fix that I guess we can use `a.weight - b.weight` 
I tried this. In case of function returning `a.weight - b.weight`  the result goes same in case of interface 'actions datagrid` goes first when expected to go last. Moreover, when one weight is undefined the result is completely unpredictable. In case of same weight the result is unpredictable in local manner (relative only to same weight items).

Reply:
@naarrek This sounds like issue related to `undefined` values, maybe we can try taking care of them as well, maybe?

```
nav.sub_items.sort((a, b) => {
        if (!a.weight)
          return 1;
        if (!b.weight)
          return -1;
        return a.weight - b.weight;
      });
```

Please let me know if that fixes the issue.